### PR TITLE
MRG: Fix onset dragging

### LINKED ISSUE #5116: raw.plot() annotations can't be dragged to extend anymore
Using master, dragging an edge from an existing annotation doesn't extend it anymore. Strangely, there must be something else going on (maybe related to a matplotlib update?), because annotation mode (i.e. pressing "a" in the raw viz window) doesn't work before commit https://github.com/mne-tools/mne-python/commit/dba99f19e24ed3e59f72bb8eaff6174c104568ca (I get an `IndexError` in `matplotlib/gridspec.py`) when the annotation window is created. Something in that commit seemed to fix that (I don't know what), but even though this got magically fixed somehow, annotations can't be dragged anymore.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #5116.

To test:
```
import mne
raw = mne.io.read_raw_fif(mne.datasets.sample.data_path() +
                          "/MEG/sample/sample_audvis_filt-0-40_raw.fif")
raw.annotations = mne.Annotations([1], [1], 'BAD_DRAG')
fig = raw.plot()
fig.canvas.key_press_event('a')  # annotation mode
```

### CODE REVIEW

**Comment by larsoner** [comment]
@cbrnr feel free to try it

**Comment by cbrnr** [comment]
There's still no filled area when dragging, is this intentional? Also, when the mouse pointer is close to the edge of an annotation, the line doesn't get thicker as before, but apparently some kind of lighter color is used to highlight the line.

Finally, the edge lines are always red, even for annotations that are e.g. green. Not sure if this was always the case.

**Comment by larsoner** [comment]
> There's still no filled area when dragging, is this intentional?

Looking at the code I don't think this was ever done (and would take a bit of work to add) so let's do it separately.

> Also, when the mouse pointer is close to the edge of an annotation, the line doesn't get thicker as before, but apparently some kind of lighter color is used to highlight the line.

I changed the appearance because I thought this was better, feel free to tweak the code (you can see how I did it in `viz/utils.py` to taste and let me know what you prefer.

**Comment by larsoner** [comment]
> Finally, the edge lines are always red, even for annotations that are e.g. green

Yes this would have been the case before, too. We might be able to change it somehow but let's do it separately because it's not clear how easy it would be.

**Comment by cbrnr** [comment]
I like the way it looks now, I'd only use matching colors and optionally implement a fill between. But as you say, this is for another PR and it is important that this gets merged ASAP. Should I create an issue to remind us or are you going to create it?

File: mne/viz/utils.py [hunk 1]
```diff
@@ -2153,7 +2155,7 @@ def _connection_line(x, fig, sourceax, targetax, y=1.,
                   clip_on=False)
 
 
-class DraggableLine:
+class DraggableLine(object):
```
Comment (cbrnr) [inline mne/viz/utils.py]: Is this really still necessary? I mean explicitly deriving from `object`?

Comment (larsoner) [reply mne/viz/utils.py]: https://stackoverflow.com/questions/4015417/python-class-inherits-object

**Comment by larsoner** [comment]
Okay @cbrnr I took the time to implement the `Poly` dragging and color matching, and fixed a bug with texts not being removed, feel free to try.

File: mne/viz/utils.py (hunk 1 above)
Comment (cbrnr) [reply mne/viz/utils.py]: Thanks - I don't even want to know, just really looking forward to 2020 :smile:.

**Comment by cbrnr** [comment]
You call this "minor fixes" 😛? This is great, thanks! I'll test it tomorrow!

**Comment by cbrnr** [comment]
Very nice! I like it! There's only one minor issue, the event colors seem to change when I close and re-open the figure. To reproduce, first create a `BAD_` event, and then create a `test` event (which is green when I create it). Close the figure, plot it again, and the `test` event is now blue.

**Comment by larsoner** [comment]
Color cycle logic fixed

**Pull request merged**